framework/src/outputs/Exodus.C

```diff
@@ -399,6 +397,23 @@ Exodus::output(const ExecFlagType & type)
 
   // Reset the mesh changed flag
   _exodus_mesh_changed = false;
+
+  // Remove empty files
+  // Because of the ability to control when different items output independently (i.e.,
+  // "execute_on_input") it is possible to create an empty file if the solve stops early, such as
+  // the case when --half-transient is used. Thus, to avoid having empty files laying around the
+  // following cleans them up. It is not possible to know if anything will be output to the file
+  // prior to attempting. A long story short, an exodus file with only info records doesn't write
+  // correctly. Deleting the file here actually doesn't fix this problem, but it seems like poor
+  // form to leave empty files around.
+  std::string current = filename();
+  if (MooseUtils::file_exists(current) && (MooseUtils::file_size(current) == 0) &&
```

Comment:
If you check for processor_id() first, this will short circuit and you will avoid potentially expensive I/O operations when running at scale.

Reply:
Good idea